src/System.Buffers/src/System/Buffers/NativeBufferBucket.cs

```diff
@@ -33,13 +34,26 @@ internal NativeBufferBucket(int elementsInBuffer, int numberOfBuffers)
             }
         }
 
-        ~NativeBufferBucket()
```

Comment:
I don't think the finalizer should be removed. I think we need both a finalizer and Dispose. The finalizer is an insurance in case dispose is not called.